### PR TITLE
chore(model/migration): employee job details and dashboard timestamps

### PR DESCRIPTION
**Pull request by Carl-Tabuso**
### 🛠 Changes

- https://github.com/POWERLANE-TEAM/emplo-hrms/pull/116/commits/1f2de17150dd4dea359751b4774cfb98c756bfdb New migration and model for employee job details. This'll make it easier to work with since there is some sort of a summary table that we can always look up for any job-related information of an employee.
- Adjust the update query builder to eloquent method of `RatingScale` and `PreEmployment` to allow logging of model events.
- Separate regular dropdown to a multi-select one. I ran into problems in announcement page wherein the dropdown would look unstyled because it is missing a `wire:ignore` directive. But upon adding it, the components that rely on it, esp the dependent dropdowns in account creation don't properly work and it's not dynamically rendering on the options anymore.
- Timestamps for both recent activity logs and announcements with publisher were added on the dashboard.
- Some attributes on `User` model, esp those which are sensitive, such as social provider ids (facebook, google), as well as those concerning auths (2fa recovery codes), etc, were exempted when logging.
- Performance details migration and model relationship were tweaked a lil bit:
   - `hr_manager` -> `secondary_approver`
   - `area_manager` -> `initial_approver`
   - `supervisor` -> _removed_
- For deeply nested relationships, i prefer using `hasOneThrough` or `hasManyThrough`. You can notice this on `Employee` model where i've written methods to join through a model, **through** the `EmployeeJobDetail` model.
- I advise running a fresh migration.
 
### ✔️ Checklist

- [x] Is `staging` the base branch of this PR?
- [x] Do your changes not reveal sensitive information, such as secrets, API keys, etc?
- [x] Are there no erroneous console logs, debuggers, or leftover code in your changes?

### CODE REVIEW

File: app/Livewire/Admin/Config/Form/PreEmployment.php [hunk 1]
```diff
@@ -21,25 +21,33 @@ class PreEmployment extends Component
     
     public function save()
     {
-        $this->validate();
-
         if ($this->editMode) {
             if (! Auth::user()->hasPermissionTo(UserPermission::CREATE_PREEMPLOYMENT_REQUIREMENTS)) {
```
Comment (C-Ivan-Bandilla) [inline app/Livewire/Admin/Config/Form/PreEmployment.php:25]: I dunno if the permission is correct here

File: app/Livewire/Admin/Dashboard/LatestAnnouncements.php [hunk 2]
```diff
@@ -39,8 +39,8 @@ public function announcements()
                     'id' => $item->announcement_id,
                     'title' => $item->announcement_title,
                     'description' => $item->announcement_description,
-                    'published_at' => $item->published_at,
-                    'modified_at' => $item->modified_at,
+                    'published_at' => $item->published_at->diffForHumans(),
+                    'modified_at' => $item->modified_at->diffForHumans(),
```
Comment (C-Ivan-Bandilla) [inline app/Livewire/Admin/Dashboard/LatestAnnouncements.php:43]: Up ing for future reference

**Comment by Carl-Tabuso** [comment]
i'll be having follow-up changes. Don't merge this yet.

**Comment by C-Ivan-Bandilla** [comment]
Is there column to save this?

https://www.figma.com/design/N0pdVwz5hw0IlNJYubxaND/Powerlane?node-id=1201-38450&node-type=frame&t=kVzrw31q3ZEenfjk-0

File: app/Livewire/Admin/Config/Form/PreEmployment.php (hunk 1 above)
Comment (Carl-Tabuso) [reply app/Livewire/Admin/Config/Form/PreEmployment.php:25]: why, isn't that supposed to be for adding a new preemployment requirements?

**Comment by C-Ivan-Bandilla** [comment]
> For supervisors, aren't there sort of way to easily know them depending on which job title an applicant applied for? Or it doesn't matter which job family, someone from accounting can have a supervisor from general-affairs?

@biellamariscotes @cristianmm1

**Comment by Carl-Tabuso** [comment]
finished

**Comment by C-Ivan-Bandilla** [comment]
I think I have outdated json right?

image


is there a new file for public\storage\utils\job-titles.json?

**Comment by Carl-Tabuso** [comment]
> I think I have outdated json right?

always slips my mind [job-titles.json](https://github.com/user-attachments/files/17900589/job-titles.json)

File: app/Livewire/Admin/Config/Form/PreEmployment.php (hunk 1 above)
Comment (Carl-Tabuso) [reply app/Livewire/Admin/Config/Form/PreEmployment.php]: swapped it

**Pull request merged**